sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1255,6 +1255,35 @@ def test_dtype_convert():
     assert_array_equal(result, y)
 
 
+def test_sparse_metric_callable():
+    def sparse_metric(x, y):  # Metric accepting sparse matrix input (only)
+
+        return min(x.nnz, y.nnz) / max(x.nnz, y.nnz)
```

Comment:
Sorry, I have to take back the LGTM. :\ This is apparently producing different results on old scipy, and tests are failing. Perhaps just assert isspmatrix(x) and return some more ordinary function of the inputs.